packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -31,7 +29,7 @@ const int kIsolateReloadBarred = 1005;
 
 /// Override `WebSocketConnector` in [context] to use a different constructor
 /// for [WebSocket]s (used by tests).
-typedef WebSocketConnector = Future<io.WebSocket> Function(String url, {io.CompressionOptions compression, @required Logger logger});
+typedef WebSocketConnector = Future<io.WebSocket> Function(String url, {io.CompressionOptions compression, Logger? logger});
```

Comment:
I don't see it, where is logger nullable?  Doesn't it need to be present in `handleError`?

Reply:
Hmm, tracing back up the usage, it looks like the only place the logger wasn't passed was in tests. Let me try to update the tests

Reply:
made this non-nullable